lua/pixelui/core/cl_images.lua

```diff
@@ -50,15 +50,27 @@ local function processQueue()
     end
 end
 
-function PIXEL.GetImgur(id, callback, _, matSettings)
-    if materials[id] then
-        callback(materials[id])
-    elseif file.Exists("pixel/" .. id .. ".png", "DATA") then
-        materials[id] = Material("../data/pixel/" .. id .. ".png", matSettings or "noclamp smooth mips")
-        callback(materials[id])
+function PIXEL.GetImage(url, callback, matSettings)
+    local urlWithoutProtocol = string.gsub(url, "https://", "")
+    urlWithoutProtocol = string.gsub(urlWithoutProtocol, "http://", "")
+
+    local fileName = url:match("[^/]+$")
+    local urlWithoutFileName = string.sub(urlWithoutProtocol, 1, url:find("/[^/]+$") - 1)
```

Comment:
Use one or the other, not both.
```suggestion
    local urlWithoutFileName = urlWithoutProtocol:sub(1, url:find("/[^/]+$") - 1)
```